src/bepasty/tests/test_app.py

```diff
@@ -24,3 +42,61 @@ def test_secret_key():
     }
     app = create_app()
     assert app.config['SECRET_KEY'] != secret_key
+
+
+class TestView(MethodView):
+    callback = None
+
+    def get(self):
+        TestView.callback()
+        return 'done'
+
+
+def prepare(callback):
+    app = create_app()
+    app.add_url_rule('/test_call', view_func=TestView.as_view('test.test_call'))
+
+    TestView.callback = staticmethod(callback)
+    client = app.test_client()
+
+    assert app.config['APP_BASE_PATH'] == Config.APP_BASE_PATH
+
+    return app, client
+
+
+def test_none(app_fixture):
+    Config.APP_BASE_PATH = None
+
+    def none_callback():
+        url = url_for('test.test_call')
+        assert url == request.path
+        url = url_for('test.test_call', _external=True)
+        assert url == 'http://localhost' + request.path
+
+    app, client = prepare(none_callback)
+
+    response = client.get('/bepasty/test_call')
+    assert response.status_code == 404
+
+    response = client.get('/test_call')
+    assert response.status_code == 200
+    assert response.data == 'done'.encode()
+
+
+def test_prefix(app_fixture):
+    Config.APP_BASE_PATH = '/bepasty'
+
+    def prefix_callback():
+        url = url_for('test.test_call')
+        assert url == Config.APP_BASE_PATH + request.path
+        url = url_for('test.test_call', _external=True)
+        assert url == 'http://localhost' + Config.APP_BASE_PATH + request.path
+
```

Comment:
same here? is APP_BASE_PATH a flask or a bepasty feature?

Reply:
APP_BASE_PATH is bepasty's PrefixMiddleware() stuff

Reply:
ok